torch/csrc/jit/codegen/cuda/executor_utils.h

```diff
@@ -113,6 +114,27 @@ class ParallelIterExtentMap {
       CompileTimeEntryType::PARALLEL_ITER_EXTENT_MAP;
 };
 
+//! Compile-time info to be cached in each FusionExecutor:
+//!  SimplifiedParallelIterExtentMap
+//!    This entry type is a simplified version of ParallelIterExtentMap.
+//!
+//!    For launch parameter binding we only need the most concrete iterdomain
+//!      in each disjoint set stored in CaParallelMap. This entry stores the
```

Comment:
I actually think the parallel map is fine here since concrete domains are selected.

Reply:
This actually makes me think again about `ParallelDimensionMap` as well. For example:

```
 auto tv0 = makeSymbolicTensor(1);
  fusion.addInput(tv0);
  auto tv1 = makeSymbolicTensor(2);
  fusion.addInput(tv1);
  auto tv2 = broadcast(tv0, {true, false});
  auto tv3 = add(tv2, tv1);
  fusion.addOutput(tv4);

  tv3->merge(0, 1);
  tv0->computeAt(tv3, -1);
  tv1->computeAt(tv3, -1);

  tv3->axis(0)->parallelize(ParallelType::BIDx);
```
```
Inputs:
  T0_g[ iS0{i1} ], float
  T1_g[ iS9{( i4 * i7 )} ], float
Outputs:
  T3_g[ iblockIdx.x7{( i4 * i1 )} ] produce_pos( 1), float

%kernel_math {
T2_l[ iS8{( 1 * i1 )} ] ca_pos( 1 ) = broadcast( T0_g[ iS0{i1} ] )
T3_g[ iblockIdx.x7{( i4 * i1 )} ] produce_pos( 1)
   = T2_l[ iS8{( 1 * i1 )} ] ca_pos( 1 )
   + T1_g[ iS9{( i4 * i7 )} ];
}
```

With this fusion, the parallel dimension map looks like:

```
blockIdx.x: gridDim.x, non-exact
blockIdx.y: unused
blockIdx.z: unused
threadIdx.x: unused
threadIdx.y: unused
threadIdx.z: unused
```

Note that `BIDx` is marked as non-exact, even though it is actually exact. This is because `ParallelDimensionMap` uses the index map so the merged axis of `T3` is not mapped with the axis of `T0`, but both of them are mapped with each other in the parallel map and are parallelized by `BIDx`.

Reply:
I thought this should be not exact since `iS8{( 1 * i1 )} != iblockIdx.x7{( i4 * i1 )}`

Reply:
I thought that was the behavior we're interested in.

Reply:
I think you meant `iS0{i1} != iblockIdx.x7{( i4 * i1 )}`. `iS8{( 1 * i1 )}` and `iblockIdx.x7{( i4 * i1 )}` are mapped in the index map. The difference comes from the forwarding, so `iS0{i1}` and `iblockIdx.x7{( i4 * i1 )}` are mapped in the parallel map but not in the index map.

Since `i1 != i4 * i1`, they appear to make `BIDx` non-exact. However, with the computeAt, indexing `T0` is always done with through `i4 * i1`. In this particular case, while `iS0{i1}` is marked as parallelized with `BIDx`, its indexing is `blockDim.x % i1`, so it never goes beyond `i1`. So, in that sense, `BIDx` is still exact, in other words, we don't need a predicate like `blockDim.x < i1`. 

This seems related to the issue we discussed before about the indexing change with and without computeAt. 

I feel this is too much intricated. I'm spending too much time to think about which maps to use. Would be great if we could come up with a single map, but not sure if that's reasonably possible.

Reply:
`iS8{( 1 * i1 )} != iblockIdx.x7{( i4 * i1 )}` should now be true in the index map, it was true that it used to map, but I explicitly changed that because it shouldn't.

This makes sense though. I forgot we don't map to the right of the compute at point in the parallel map. So really this is what parallel type is being used to index into the tensor. In that instance then yes, the parallel map is enough to get the unique entries of parallelization bound into the problem.